csp/models.py

```diff
@@ -49,25 +48,24 @@ def strip_uri(cls, uri: str) -> str:
         """
         return strip_query(uri)[:200] if uri else ""
 
-    @root_validator
-    def validate_directives(cls, values: PydanticValues) -> PydanticValues:
+    @model_validator(mode="after")
+    def validate_directives(self) -> ReportData:
         """Ensure that we have either effective_directive or violated_directive."""
-        if values.get("effective_directive", ""):
-            return values
+        if self.effective_directive:
+            return self
         # if effective_directive is empty, but violated_directive is not,
-        # then udpate the former with the latter.
-        if violated_directive := values.get("violated_directive", ""):
+        # then update the former with the latter.
+        if self.violated_directive:
             logger.debug(
                 "'effective_directive' missing - using 'violated_directive' attr."
             )
-            values["effective_directive"] = violated_directive
-            return values
+            self.effective_directive = self.violated_directive
+            return self
         raise ValueError(
             "Either 'effective_directive' or 'violated_directive' must be present."
         )
 
-    class Config:
-        allow_population_by_field_name = True
+    model_config = ConfigDict(populate_by_name=True, coerce_numbers_to_str=True)
```

Comment:
`coerce_numbers_to_str=True` was added to allow `status_code` to be an integer or a string, which was the behaviour before this change. 
This is because Pydantic 1.x.x would coerce numbers to strings by default, whereas Pydantic 2.x.x will not do this by default.

I'm not sure if we want to keep this behaviour or not though.